quinn-proto/src/connection/mod.rs

```diff
@@ -475,6 +475,7 @@ impl Connection {
         // Position in `buf` of the first byte of the current UDP datagram. When coalescing QUIC
         // packets, this can be earlier than the start of the current QUIC packet.
         let mut datagram_start = 0;
+        let mut segment_size = self.path.current_mtu();
```

Comment:
I suspect this might be easier to review if you split the mere naming of `self.path.current_mtu()` (where used as a `segment_size`) into a separate commit. Also, since we're naming this, based on this diff it looks like the natural type for this might be `usize`, not `u16`?